src/Symfony/Component/Message/HandlerLocator.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Message;
+
+use Symfony\Component\Message\Exception\NoHandlerForMessageException;
+use Symfony\Component\Message\Handler\MessageHandlerCollection;
+
+/**
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+class HandlerLocator implements HandlerLocatorInterface
```

Comment:
@sroze: HandlerLocator should be able to lazy load the handlers. Now the question is, can I change this class to have ServiceLocator injected into the constructor, or should a new class be created and set as default in framework bundle.

Reply:
+1 for a PSR-11 implem. (IMHO fits core/component)

Reply:
For the records, @mvrhov started https://github.com/sroze/symfony/pull/3 👏